### PR TITLE
Add UI implementation for CRD workflow

### PR DESCRIPTION
**Pull request by Thiyan**

### CODE REVIEW

File: demo-ehr-app/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# React + TypeScript + Vite
```
Comment (sameeragunarathne) [inline demo-ehr-app/README.md]: README should updated by including the details about the demo-ehr-app and how to set it up.

File: demo-ehr-app/package.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{
```
Comment (sameeragunarathne) [inline demo-ehr-app/package.json:1]: Do we need to commit this file? only package.json is enough?

File: demo-ehr-app/src/App.css [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#root {
```
Comment (sameeragunarathne) [inline demo-ehr-app/src/App.css:19]: Let's add apache licence for new source files

File: demo-ehr-app/src/components/cds_dev_portal.tsx [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+  const location = useLocation();
+  const lastUrlSegment = location.pathname.split("/").pop();
+
+  // const showResponse =
+  //   lastUrlSegment === "coverage-preview" || lastUrlSegment === "coverage";
+
+  console.log("Last URL Segment: ", lastUrlSegment);
+
+  // const cdsRequest = {
```
Comment (sameeragunarathne) [inline demo-ehr-app/src/components/cds_dev_portal.tsx]: Remove commented out codes, cleanup everywhere

File: demo-ehr-app/src/redux/store.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import cdsRequestSlice from './cdsRequestSlice';
+import cdsResponseSlice from './cdsResponseSlice';
+
+export default configureStore({
+  reducer: {
+    patient: patientReducer,
+    cdsRequest: cdsRequestSlice,
+    cdsResponse: cdsResponseSlice,
+  },
+});
```
Comment (sameeragunarathne) [inline demo-ehr-app/src/redux/store.ts]: Add EoF. check other files as well

File: demo-ehr-app/package.json (hunk 2 above)
Comment (Thiyan) [reply demo-ehr-app/package.json:1]: Yes

File: demo-ehr-app/index.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
Comment (isuruh15) [inline demo-ehr-app/index.html]: Shall we have a proper title here?

File: demo-ehr-app/src/pages/MedicalImaging.tsx [hunk 7]
```diff
@@ -0,0 +1,460 @@
...
+
+  const validateError = (
+    <>
+      <Alert severity="error" sx={{ marginLeft: 1 }}>
+        This is an mandotary field.
+      </Alert>
+    </>
+  );
+
+  const [patientId] = useState("PA2347");
```
Comment (isuruh15) [inline demo-ehr-app/src/pages/MedicalImaging.tsx:149]: Don't we need to keep the IDs in `data.ts` file?

File: demo-ehr-app/src/pages/MedicalImaging.tsx [hunk 8]
```diff
@@ -0,0 +1,460 @@
...
+      practionerId,
+      selectedDate?.format("YYYY-MM-DD"),
+      selectedSlots
+    );
+
+    setCdsCards([]);
+    setEnableSubmitButton(false);
+    axios
+      .post<CdsResponse>(
+        "https://c32618cf-389d-44f1-93ee-b67a3468aae3-dev.e1-us-east-azure.choreoapis.dev/cmsdemosetups/cds-server/v1.0/cds-services/radiology-order",
```
Comment (isuruh15) [inline demo-ehr-app/src/pages/MedicalImaging.tsx]: shall we take this to a config file?

Comment (isuruh15) [reply demo-ehr-app/src/pages/MedicalImaging.tsx]: and check the other places as well

File: demo-ehr-app/src/utils/cds_response_constructor.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+      "cards": []
+    }
+  }
+
+  if (hook === "cds-services"){
+    return CDS_SERVICE_SAMPLE_RESPONSE;
+  }
+
+  // make an api call to localhost:8080/cds-services
+  const response = await fetch("http://localhost:8080/cds-services/" + hook, {
```
Comment (isuruh15) [inline demo-ehr-app/src/utils/cds_response_constructor.ts:35]: Is this valid for Choreo deployments as well?

File: demo-ehr-app/src/redux/store.ts (hunk 5 above)
Comment (Thiyan) [reply demo-ehr-app/src/redux/store.ts]: Fixed. https://github.com/wso2/solutions-healthcare-demos/pull/2/commits/cbd3537da387e27bd4a9a2c3dce62aad2f02ade3

File: demo-ehr-app/src/components/cds_dev_portal.tsx (hunk 4 above)
Comment (Thiyan) [reply demo-ehr-app/src/components/cds_dev_portal.tsx]: Fixed. https://github.com/wso2/solutions-healthcare-demos/pull/2/commits/0c39df26a886dfb366e9e70252f92ee7049deee2

File: demo-ehr-app/src/utils/cds_response_constructor.ts (hunk 9 above)
Comment (Thiyan) [reply demo-ehr-app/src/utils/cds_response_constructor.ts:35]: Joel implemented these, but they are irrelevant to the current implementation. We need a dedicated effort to refactor.

File: demo-ehr-app/src/pages/MedicalImaging.tsx (hunk 7 above)
Comment (Thiyan) [reply demo-ehr-app/src/pages/MedicalImaging.tsx:149]: In the data.ts file, we have templates or placeholders to parse the dynamic data like patientId and practinionerId in the request payload.

File: demo-ehr-app/src/pages/MedicalImaging.tsx (hunk 8 above)
Comment (Thiyan) [reply demo-ehr-app/src/pages/MedicalImaging.tsx]: Fixed. https://github.com/wso2/solutions-healthcare-demos/pull/2/commits/afd92c12223917bcd39b551d212f4129aae9b4bc

File: demo-ehr-app/index.html (hunk 6 above)
Comment (Thiyan) [reply demo-ehr-app/index.html]: Fixed. https://github.com/wso2/solutions-healthcare-demos/pull/2/commits/afd92c12223917bcd39b551d212f4129aae9b4bc

File: demo-ehr-app/README.md (hunk 1 above)
Comment (Thiyan) [reply demo-ehr-app/README.md]: Updated. https://github.com/wso2/solutions-healthcare-demos/pull/2/commits/2c489a67c1aa2e1022389842e01410f0410ea060

File: demo-ehr-app/src/App.css (hunk 3 above)
Comment (Thiyan) [reply demo-ehr-app/src/App.css:19]: Added. https://github.com/wso2/solutions-healthcare-demos/pull/2/commits/6e044a5fd78a9e997fccca0f49b0fe94a76abb9d

File: demo-ehr-app/src/utils/cds_response_constructor.ts (hunk 9 above)
Comment (Thiyan) [reply demo-ehr-app/src/utils/cds_response_constructor.ts:35]: https://github.com/wso2-enterprise/open-healthcare/issues/1675

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**